xbmc/music/Album.cpp

```diff
@@ -56,6 +57,33 @@ CAlbum::CAlbum(const CFileItem& item)
 
   if (!tag.GetMusicBrainzAlbumArtistID().empty())
   { // have musicbrainz artist info, so use it
+    // Establish tag consistency - do the number of musicbrainz ids and number of names in hints or artist match
+    if (tag.GetMusicBrainzAlbumArtistID().size() != musicBrainAlbumArtistHints.size() &&
+      tag.GetMusicBrainzAlbumArtistID().size() != tag.GetAlbumArtist().size())
+    {
+      // Tags mis-match - report it and then try to fix
+      CLog::Log(LOGDEBUG, "Mis-match in song file albumartist tags: %i mbid %i name album: %s",
+        tag.GetMusicBrainzAlbumArtistID().size(), tag.GetAlbumArtist().size(), strAlbum);
```

Comment:
Should be strAlbum.c_str() and shouldn't use %i as you're passing size_t, can't remember which format is correct